### ISSUE TITLE
squeeze on 1x1 tensor causes error in gradients while view(-1) doesn't

### ISSUE THREAD

**Issue by stefdoerr** [pytorch/pytorch#4778]
```python
import torch
from torch.autograd import grad, Variable

w = Variable(torch.ones([3, 1])).cuda()
xyz = Variable(torch.FloatTensor([[0, 1.3, 2], ]), requires_grad=True).cuda()
res = torch.matmul(xyz, w)

# Works
torch.autograd.grad(res.view(-1), xyz, torch.ones_like(res.squeeze()), retain_graph=True)
# Doesn't
torch.autograd.grad(res.squeeze(), xyz, torch.ones_like(res.squeeze()), retain_graph=True)
```

**Comment by zou3519** [comment]
Looking into this. It looks like `squeeze`'s backwards might not take into account this case.

**Comment by stefdoerr** [comment]
Thanks so much for fixing it on such short order! 😄
